backend/consensus/worker_service.py

```
@@ -153,11 +173,25 @@ def get_session():
+    global worker, worker_task
 
     if worker is None:
         return {"status": "initializing", "worker_id": None}
 
+    # Check if worker task has died
+    if worker_task and worker_task.done():
+        # Worker task finished unexpectedly - this is a failure
+        try:
+            # This will re-raise any exception from the task
+            worker_task.result()
+        except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)